Taarafo.Core.Tests.Unit/Services/Foundations/Comments/CommentServiceTests.Validations.RetrieveById.cs

```diff
@@ -0,0 +1,94 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Moq;
+using Taarafo.Core.Models.Comments;
+using Taarafo.Core.Models.Comments.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Comments
+{
+    public partial class CommentServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRetrieveByIdIfIdIsInvalidAndLogItAsync()
+        {
+            // given
+            var invalidCommentId = Guid.Empty;
+
+            var invalidCommentException =
+                new InvalidCommentException();
+
+            invalidCommentException.AddData(
+                key: nameof(Comment.Id),
+                values: "Id is required");
+
+            var expectedCommentValidationException = new
+                CommentValidationException(invalidCommentException);
+
+            // when
+            ValueTask<Comment> retrieveCommentByIdTask =
+                this.commentService.RetrieveCommentByIdAsync(invalidCommentId);
+
+            // then
+            await Assert.ThrowsAsync<CommentValidationException>(() =>
+                retrieveCommentByIdTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedCommentValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectCommentByIdAsync(It.IsAny<Guid>()),
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowNotFoundExceptionOnRetrieveByIdIfCommentIsNotFoundAndLogItAsync()
```

Comment:
This test should be under Validations.RetrieveById.cs file.
@hassanhabib Please correct if I am wrong.

Reply:
![image](https://user-images.githubusercontent.com/797750/140816263-1fe520eb-0a3d-4ed2-81ad-b29a12349199.png)
@ShriHumrudha the filename you suggest looks the same as its location?

Reply:
My bad. Thank you for confirming.